learning_table_tennis_from_scratch/hysr_one_ball.py

```diff
@@ -90,6 +90,10 @@ def from_json(cls, jsonpath):
                 raise ValueError(
                     "failed to find the attribute {} " "in {}".format(s, jsonpath)
                 )
+
+        # convert paths to Path objects and expand '~'
+        instance.pam_config_file = pathlib.Path(instance.pam_config_file).expanduser()
```

Comment:
@vincentberenz I had an error because `pam_config_file` was used in a place where the `~` had not been replaced.  I thought it might be best to replace it directly here at the lowest level, so code using the HysrOneBallConfig class doesn't need to worry about it anymore.

I wasn't sure if I should keep it as Path or convert to string here.  I went with Path now (meaning that it needs to be converted to string in a few places), let me know if you prefer it the other way round, then I'll change it.

Reply:
Anything that works is ok for me :) I hope to finish the clean refactor of HYSR soon !